controllers/instancegroup_controller.go

```diff
@@ -184,34 +179,61 @@ func (r *InstanceGroupReconciler) Reconcile(ctxt context.Context, req ctrl.Reque
 
 	if err = input.InstanceGroup.Validate(); err != nil {
 		ctx.SetState(v1alpha1.ReconcileErr)
-		r.UpdateStatus(input.InstanceGroup)
+		r.UpdateStatus(input.InstanceGroup, status)
 		return ctrl.Result{}, errors.Wrapf(err, "provisioner %v reconcile failed", provisionerKind)
 	}
 
 	if err = HandleReconcileRequest(ctx); err != nil {
 		ctx.SetState(v1alpha1.ReconcileErr)
-		r.UpdateStatus(input.InstanceGroup)
+		r.UpdateStatus(input.InstanceGroup, status)
 		return ctrl.Result{}, errors.Wrapf(err, "provisioner %v reconcile failed", provisionerKind)
 	}
 
 	if provisioners.IsRetryable(input.InstanceGroup) {
 		r.Log.Info("reconcile event ended with requeue", "instancegroup", req.NamespacedName, "provisioner", provisionerKind)
-		r.UpdateStatus(input.InstanceGroup)
+		r.UpdateStatus(input.InstanceGroup, status)
 		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
-	r.UpdateStatus(input.InstanceGroup)
+	r.UpdateStatus(input.InstanceGroup, status)
 	r.Finalize(instanceGroup)
 	return ctrl.Result{}, nil
 }
 
-func (r *InstanceGroupReconciler) UpdateStatus(ig *v1alpha1.InstanceGroup) {
-	r.Log.Info("updating resource status", "instancegroup", ig.NamespacedName())
-	if err := r.Status().Update(context.Background(), ig); err != nil {
+func (r *InstanceGroupReconciler) UpdateStatus(instanceGroup *v1alpha1.InstanceGroup, status *v1alpha1.InstanceGroupStatus) {
+	r.Log.Info("updating resource status", "instancegroup", instanceGroup.NamespacedName())
+	newStatus := instanceGroup.GetStatus()
+	if reflect.DeepEqual(newStatus, status) {
+		r.Log.Info("skipping update due to no changes", "instancegroup", instanceGroup.NamespacedName())
+		return
+	}
+	if err := r.Status().Update(context.Background(), instanceGroup); err != nil {
 		// avoid error if object already deleted
 		if kubeprovider.IsStorageError(err) {
 			return
 		}
-		r.Log.Info("failed to update status", "error", err, "instancegroup", ig.NamespacedName())
+		r.Log.Info("failed to update status", "error", err, "instancegroup", instanceGroup.NamespacedName())
+	}
+}
+
+func (r *InstanceGroupReconciler) IsNamespaceExcluded(instanceGroup *v1alpha1.InstanceGroup) bool {
+	if reflect.DeepEqual(r.ConfigMap, &corev1.ConfigMap{}) {
```

Comment:
Is this check necessary? Seems like it's outside of the scope of what this function should need to do (and, in fact, is checked already before calling this function). Semantically, I'd assume that IsNamespaceExcluded is a separate check from whether or not there is actually a managed configuration

Reply:
Yep, this is probably redundant